src/centered-on-asteroid/model_def.py

```diff
@@ -0,0 +1,191 @@
+import torch
+import torch.nn as nn
+from typing import Tuple, List
+
+
+class DynamicCNN(nn.Module):
+    def __init__(
+        self,
+        image_shape: Tuple[int, int] = (30, 30),
+        num_conv_blocks: int = 2,
+        filters_list: List[int] = [16, 32],
+        kernel_size: int = 3,
+        stride: int = 1,
+        padding: int = 1,
+        output_size: int = 1,
+    ):
+        super(DynamicCNN, self).__init__()
+        self.image_shape = image_shape
+
+        self.num_conv_blocks = num_conv_blocks
+        self.filters_list = filters_list
+        self.kernel_size = kernel_size
+        self.stride = stride
+        self.padding = padding
+
+        self.maxpool_kernel_size = 2
+        self.maxpool_stride = 2
+
+        self.conv_blocks = nn.ModuleList(
+            [nn.Conv2d(1, filters_list[0], kernel_size, stride, padding)]
+        )
+        self.conv_blocks.extend(
+            [
+                nn.Conv2d(
+                    filters_list[i], filters_list[i + 1], kernel_size, stride, padding
+                )
+                for i in range(num_conv_blocks - 1)
+            ]
+        )
+
+        # We only need one ReLU layer and one maxpool layer, as they don't learn any parameters
+        self.relu = nn.ReLU()
+
+        self.maxpool = nn.MaxPool2d(
+            kernel_size=self.maxpool_kernel_size, stride=self.maxpool_stride
+        )
+        output_size_after_max_pools = (
+            image_shape[0] // self.maxpool_kernel_size**self.num_conv_blocks,
```

Comment:
I think this needs to be self.maxpool_stride not kernel size, as the decrease in the size of the image is governed by the stride

Reply:
Oh good catch will change that now!